hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,290 @@ title: "OTP Controller Technical Specification"
+These calibration values are programmed into the CREATOR_SW_CFG partition, as they are non-secret values meant to be read out by software and programmed into respective peripherals.
+
+Early on during this stage, the various tokens are also programmed into the secret partitions and harvested by the silicon creator.
+
+### Provisioning
+
+During this stage, the device is provisioned with the final firmware and a "unique" seed or identity.
+The secret partitions are populated with root secrets and keys that are critical to establishing the device identity.
+
+As part of injecting the final firmware, the stock-keeping-unit-specific hardware and software configurations are also programmed.
+
+### Life Cycle Partition
+
+The life cycle partition is active throughout all stages and hence it is the **ONLY** partition that cannot be locked.
+After the device finishes provisioning and goes into production, it must retain the ability to transition back to RMA in case of unexpected failures.
+
+In order to support this transition, the life-cycle state and counters must always be update-able (**TODO: add link to life-cycle docs for more info**).
+
+## Locking a Partition
+
+Write access to a partition can be permanently locked when software determines it will no longer make any updates to that partition.
+To lock, an integrity constant is calculated and programmed alongside the other data of that partition.
+The size of that integrity constant depends on the partition size granule, and is either 32bit or 64bit (see also [Direct Access Memory Map]({{< relref "#direct-access-memory-map" >}})).
+
+Once the "integrity digest" is non-zero, no further updates are allowed.
+If the partition is secret, software is in addition no longer able to read its contents (see [Secret Partition description]({{< relref "#secret-vs-nonsecret-partitions" >}})).
+
+Note however, in all partitions, the digest itself is **ALWAYS** readable.
+This gives software an opportunity to confirm that the locking operation has proceeded correctly, and if not, scrap the part immediately.
+
+Calculation of the integrity digest depends on whether the partition requires periodic background verification.
+
+### Software Configuration Partitions
+
+The software configuration partitions are used as non-volatile storage for flags, configuration and calibration data.
+As such, the contents of this partition are usually consumed once as part of code execution, or moved to another storage compartment somewhere in the design.
+For example, the clock calibration values and the LDO calibration values are programmed to the analog sensor top (AST) at startup.
+
+As such, it is not necessary to check periodically at the OTP source.
+Instead, software can simply check as part of secure boot and take other measures when these values are programmed into peripherals.
+
+For this partition it is thus the responsibility of software to calculate the integrity digest and program it into the OTP.
```

Comment:
in all cases the digest serves as the write-lock "switch". the only difference in case of SW_CFG partitions is that it is up to SW to calculate, program and verify the digest.

Reply:
@cdgori
@tjaychen